### ISSUE TITLE
Tests for visibilitychange event during unload

### ISSUE THREAD

**Issue by igrigorik** [w3c/page-visibility#21]
Current [W3C test suite](http://w3c-test.org/page-visibility/) does not cover this case. 

We have a (behind a flag) CL to enable this in Chrome (https://codereview.chromium.org/1778753003), but it would be nice to have a shared test suite to make sure that all the browsers conform and have same behavior.

**Comment by igrigorik** [comment]
@bzbarsky does Gecko have any tests for visibilitychange+unload? I did a quick scan through the code, didn't spot anything.

**Comment by bzbarsky** [comment]
I'm not sure offhand; I don't see anything obvious.

**Comment by igrigorik** [comment]
@plehegar any luck with this one? Anything we can do to help?

**Comment by plehegar** [comment]
Waiting on @igrigorik to confirm the test is proper.

**Comment by igrigorik** [comment]
@plehegar does the test still pass for you in Chrome stable? It shouldn't... What platform and Chrome version are you on?

On my machine (OSX):
- Chrome 50.0.2661.86 (64-bit): test fails in stable, as expected.
- Chrome Canary, with chrome://flags/#enable-experimental-web-platform-features enabled, the test passes. With the flag disabled, the test fails... As expected.

Do you get same results?

p.s. test page: http://w3c-test.org/submissions/2874/page-visibility/unload.html

**Comment by plehegar** [comment]
With 50.0.2661.87 on Windows 10 or 50.0.2661.86 on Linux, the test passes, independently of the state of chrome://flags/#enable-experimental-web-platform-features , as long as you allow popup, and thus using http://w3c-test.org/submissions/2874/page-visibility/unload.html . I don't have an OSX machine handy at the moment but I'll try tomorrow.

**Comment by plehegar** [comment]
Confirmed that the test fails on OSX with 50.0.2661.86. So, we do have a diff between the OS versions of Chrome here. Note that test passes on Firefox/OSX.

Why do you think that the test shouldn't pass btw?

**Comment by igrigorik** [comment]
@plehegar well, I guess you _could_ say that everything is working as we want it to. My confusion is around why the test is passing without the flag, since without it we wouldn't be triggering the [new path](https://codereview.chromium.org/1778753003/diff/200001/third_party/WebKit/Source/core/dom/Document.cpp?context=10&column_width=80&tab_spaces=8) in Chrome.. but, I guess, that's more of a Chromium question/thing to investigate. We don't have to block the tests on this.

Speaking of which, we also have this test: https://codereview.chromium.org/1778753003/diff/200001/third_party/WebKit/LayoutTests/fast/events/page-visibility-iframe-unload.html?context=10&column_width=80&tab_spaces=8

**Comment by plehegar** [comment]
I'll update the PR and add the new test. Thanks for the pointer!
